calc.go

```diff
@@ -46,5 +46,11 @@ func calcMoveDamage(weather Weather, user, receiver *Pokemon, move *Move) (damag
 		}
 	}
 	damage = uint((((levelEffect * movePower * attack / defense) / 50) + 2) * modifier)
+	elementalEffect := GetElementalEffect(move.Type(), receiver.EffectiveType())
```

Comment:
I'm not sure exactly how the order works in the damage calculation, but adding this multiplier after truncating to `uint` seems inconsistent. I feel like `modifier`, which includes all values such as STAB and type effectiveness should be consolidated before applying it to the damage formula.

Reply:
I want to refactor the damage calculation to remove all the casts to `float64`, because they cause tests to be extremely flakey.

Since all the modifier multipliers come after the initial damage calculation, doing a bitshift like this is functionally equivalent. This implementation also feels more clean because `NoEffect` matchups don't require a special case.

If anything, it's more consistent because it's not a working with a float64.